src/cascadia/TerminalSettingsModel/ActionMapSerialization.cpp

```diff
@@ -156,4 +158,12 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
 
         return keybindingsList;
     }
+
+    void ActionMap::LogSettingChanges(std::set<std::string>& changes, const std::string_view& context) const
+    {
+        for (const auto& setting : _changeLog)

```

Comment:
Currently `_changeLog` does not contain any keybinding changes (i.e. if the user adds additional keybindings to the `keybindings` array but not any new actions). Is this something we want to track? This would require additions to the `_changeLog` whenever we call `_KeyMap.insert_or_assign` (which happens once in this file and once in `ActionMap.cpp`)

Reply:
Hmm... userDefaults.json already includes these keybindings:
```js
{ "id": "Terminal.CopyToClipboard", "keys": "ctrl+c" },
{ "id": "Terminal.PasteFromClipboard", "keys": "ctrl+v" },
{ "id": "Terminal.DuplicatePaneAuto", "keys": "alt+shift+d" }
```
So we'd get a hit off of every new user. I could specifically exclude those 3 though (the specific combination of `id=Terminal.CopyToClipboard` and `keys=ctrl+c` along with the other two). And that would track how many users have added _new_ key bindings.

Sidebar: I should probably exclude `copyOnSelect=false` and `copyFormatting=false` too since they're already in userDefaults, to avoid any false positives there.

I'll work on adding this all in 😊